src/org/jitsi/impl/neomedia/transform/OriginalHeaderBlockTransformEngine.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright @ 2015 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.impl.neomedia.transform;
+
+import org.jitsi.impl.neomedia.*;
+import org.jitsi.util.*;
+
+/**
+ * Appends an Original Header Block packet extension to incoming packets.
+ * Note that we currently do NOT follow the PERC format, but rather an extended
+ * backward compatible format.
+ * {@see "https://tools.ietf.org/html/draft-ietf-perc-double-02"}
+ *
+ * Specifically the format the we currently append is
+ * <pre>{@code
+ *  0                   1                   2                   3
+ *  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+ *  +-------------+---------------+-------------------------------+
+ *  |  id  | len  |R|     PT      |        Sequence Number        |
+ *  +-------------+---------------+-------------------------------+
+ *  |                         Timestamp                           |
+ *  +-------------------------------------------------------------+
+ *  |                            SSRC                             |
+ *  +-------------------------------------------------------------+
+ *
+ * }</pre>
+ *
+ * @author Boris Grozev
+ */
+public class OriginalHeaderBlockTransformEngine
+    extends SinglePacketTransformerAdapter
+    implements TransformEngine
+{
+    /**
+     * The <tt>Logger</tt> used by the
+     * {@link OriginalHeaderBlockTransformEngine} class and its instances.
+     */
+    private static final Logger logger
+        = Logger.getLogger(OriginalHeaderBlockTransformEngine.class);
+
+    /**
+     * The ID of the OHB RTP header extension, or -1 if it is not enabled.
+     */
+    private byte extensionID = 8;
+
+    /**
+     * Initializes a new {@link OriginalHeaderBlockTransformEngine} instance.
+     */
+    public OriginalHeaderBlockTransformEngine()
+    {
+        super(RTPPacketPredicate.INSTANCE);
+    }
+
+    /**
+     * Implements {@link SinglePacketTransformer#reverseTransform(RawPacket)}.
+     */
+    @Override
+    public RawPacket reverseTransform(RawPacket pkt)
+    {
+        if (extensionID != -1)
+        {
+            // TODO: check if an OHB ext already exists.
```

Comment:
Actually it wouldn't be a simple addition, because the semantics of an existing OHB extension are somewhat complicated (if one exists then the bridge is allowed to modify extensions which come after it, and not extensions which come before it).
https://tools.ietf.org/html/draft-ietf-perc-double-02#section-5.2